src/Features/Core/Portable/SimplifyInterpolation/Helpers.cs

```diff
@@ -50,32 +63,38 @@ private static IOperation Unwrap(IOperation expression)
         }
 
         private static void UnwrapFormatString(
-            IOperation expression, out IOperation unwrapped, out string formatString)
+            ref IOperation? expression,
+            out string? formatString,
+            List<TextSpan> unnecessarySpans)
         {
-            unwrapped = null;
             formatString = null;
 
             if (expression is IInvocationOperation { TargetMethod: { Name: nameof(ToString) } } invocation)
             {
                 if (invocation.Arguments.Length == 1 &&
-                    invocation.Arguments[0].Value.ConstantValue is { HasValue: true, Value: string format })
+                    invocation.Arguments[0].Value is { ConstantValue: { HasValue: true, Value: string format } } argumentValue)
                 {
-                    unwrapped = invocation.Instance;
+                    expression = invocation.Instance;
                     formatString = format;
+
+                    unnecessarySpans.AddRange(invocation.Syntax.Span
+                        .Subtract(invocation.Instance.Syntax.FullSpan)
+                        .Subtract(argumentValue.Syntax.FullSpan));
                     return;
                 }
 
                 if (invocation.Arguments.Length == 0)
                 {
-                    unwrapped = invocation.Instance;
+                    expression = invocation.Instance;
                     formatString = "";
+                    unnecessarySpans.Add(invocation.Syntax.Span);
                     return;
                 }
             }
         }
 
         private static void UnwrapAlignmentPadding<TExpressionSyntax>(
-            IOperation expression, out IOperation unwrapped, out TExpressionSyntax alignment, out bool negate)
```

Comment:
consider going back to the 'expression/unwrapped' form.  I believe my branch shows a good way for handling this.

Reply:
I planned to do so as part of the 'Rebase' todo item